stix_shifter_modules/gcp_chronicle/stix_transmission/results_connector.py

```diff
@@ -103,19 +139,25 @@ def create_results_connection(self, search_id, offset, length):
             ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
 
         finally:
-            if 'code' in response_dict:
-                if response_dict['code'] not in (1010, 1015):
-                    self.api_client.delete_search(search_id)
-            else:
-                self.api_client.delete_search(search_id)
+            try:
+                if not return_obj.get('data') or result_count >= self.api_client.result_limit or \
+                        response_dict.get('message') or not response_text.get('nextPageToken'):
```

Comment:
can you also put a debug log here about deleting the search?

Reply:
Updated the code with logger.debug when deletion is done in results_connector.py